### ISSUE TITLE
Area/Bar stacking implementation

### ISSUE THREAD

**Issue by stas-sl** [holoviz/holoviews#5235]
This is related to https://github.com/holoviz/hvplot/issues/551, but I think it is a more broad issue. For bar charts specifying columns order works as expected (except reversing order), but for area chart this order is not respected.

```python
df = pd.DataFrame({'a': [2, 3, 1], 'b': [1, 2, 3], 'c': [3, 2, 1]})
df.hvplot.bar(y=['b', 'c', 'a'], stacked=True)
```
<img width="793" alt="image" src="https://user-images.githubusercontent.com/4602302/158224616-cabbfb8a-9443-4de5-a3f2-cfc1afeefde4.png">

```python
df = pd.DataFrame({'a': [2, 3, 1], 'b': [1, 2, 3], 'c': [3, 2, 1]})
df.hvplot.area(y=['b', 'c', 'a'], stacked=True)
```

<img width="795" alt="image" src="https://user-images.githubusercontent.com/4602302/158224700-1de48b60-b0f6-4ceb-8407-814189bb1ec2.png">

I believe this is due rather different implementations. For area it goes inside `Area.stack` and has `df.groupby(level=levels)` which causes sorting implicitly:
https://github.com/holoviz/holoviews/blob/b6ff2d14ada3833c5f2f799f10c1ea084a265150/holoviews/element/chart.py#L237-L264

Whereas for bar charts it is handled on plotting level in `BarPlot.get_stack`:
https://github.com/holoviz/holoviews/blob/b6ff2d14ada3833c5f2f799f10c1ea084a265150/holoviews/plotting/bokeh/chart.py#L828-L848

Also I'm not sure why not to just use ad hoc bokeh methods [vbar_stack](https://docs.bokeh.org/en/latest/docs/gallery/bar_stacked.html) and [varea_stack](https://docs.bokeh.org/en/latest/docs/gallery/stacked_area.html) for this.

**Comment by jbednar** [comment]
Would adding `sort=False` to the groupby call be enough to fix it?

**Comment by stas-sl** [comment]
Yes, it indeed solves the issue! I didn't know there is such option. 

```python
df = pd.DataFrame({'a': [2, 3, 1], 'b': [1, 2, 3], 'c': [3, 2, 1]})
df.hvplot.area(y=['b', 'c', 'a'], stacked=True)
```

<img width="456" alt="image" src="https://user-images.githubusercontent.com/4602302/158424892-09861dac-b544-4edf-96da-b97f74d8383d.png">

Though on a higher level maybe it worth considering unifying both these charts? Ideally it seems reasonable to extract this functionality in a common mixin and use bokeh methods, or at least to have it on the same level, whereas currently bar chart is stacked on plotting/bokeh level, while area chart is stacked on a more abstract level before specific backend.

**Comment by jbednar** [comment]
Happy to consider a PR either just making this tiny `sort=False` fix or making things more consistent, which seems like more work.
